src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -267,20 +267,22 @@ await listener.AfterExecutionAsync(context.UserContext, context.CancellationToke
             IEnumerable<IDocumentExecutionListener> listeners,
             bool throwOnUnhandledException)
         {
-            var context = new ExecutionContext();
-            context.Document = document;
-            context.Schema = schema;
-            context.RootValue = root;
-            context.UserContext = userContext;
-
-            context.Operation = operation;
-            context.Variables = GetVariableValues(document, schema, operation?.Variables, inputs);
-            context.Fragments = document.Fragments;
-            context.CancellationToken = cancellationToken;
-
-            context.Metrics = metrics;
-            context.Listeners = listeners;
-            context.ThrowOnUnhandledException = throwOnUnhandledException;
+            var context = new ExecutionContext
+            {
+                Document = document,
+                Schema = schema,
+                RootValue = root,
+                UserContext = userContext,
+
+                Operation = operation,
+                Variables = GetVariableValues(document, schema, operation?.Variables, inputs),
```

Comment:
I prefer not to call methods in object constructor syntax since if the method throws an error you do not get what line number caused the error in the stack trace, it just points to the constructor.

Reply:
I rebased onto master and moved potential exceptions out of the object initializer.